programs/ido-pool/src/lib.rs

```diff
@@ -81,20 +86,42 @@ pub mod ido_pool {
         }
         let pool_account = &mut ctx.accounts.pool_account;
         pool_account.start_ido_ts = start_ido_ts;
-        pool_account.end_deposits_ts = end_deposits_ts;
         pool_account.end_ido_ts = end_ido_ts;
         pool_account.withdraw_melon_ts = withdraw_melon_ts;
         Ok(())
     }
 
-    #[access_control(unrestricted_phase(&ctx))]
+    pub fn modify_max_usdc_tokens(
+        ctx: Context<ModifyMaxUsdcTokens>,
+        max_usdc_tokens: u64,
+    ) -> Result<()> {
+        if max_usdc_tokens == 0 {
+            return Err(ErrorCode::InvalidParam.into());
+        }
+        if Pubkey::from_str(ALLOWED_DEPLOYER).unwrap() != *ctx.accounts.payer.to_account_info().key
+        {
+            return Err(ErrorCode::InvalidParam.into());
+        }
+        let pool_account = &mut ctx.accounts.pool_account;
+        pool_account.max_usdc_tokens = max_usdc_tokens;
+        Ok(())
+    }
+
+    #[access_control(unrestricted_phase(&ctx.accounts.pool_account))]
     pub fn exchange_usdc_for_redeemable(
         ctx: Context<ExchangeUsdcForRedeemable>,
         amount: u64,
     ) -> Result<()> {
         if amount == 0 {
             return Err(ErrorCode::InvalidParam.into());
         }
+
+        // Determine exchangeable usdc amount for user.
+        let amount = cmp::min(
+            amount,
+            ctx.accounts.pool_account.max_usdc_tokens - ctx.accounts.pool_account.num_usdc_tokens,
```

Comment:
raised by soteria autoscan: possible overflow from subtraction

Reply:
In the `#[access_control(unrestricted_phase(&ctx.accounts.pool_account))]`, it is checking the max_usdc_tokens and num_usdc_tokens like this

if !(pool_account.num_usdc_tokens < pool_account.max_usdc_tokens) {
        return Err(ErrorCode::ExceedUsdc.into());
    }